benchmark/test/test_driver.py

```diff
@@ -0,0 +1,152 @@
+from itertools import count, repeat
+
+from zope.interface import implementer
+
+from twisted.internet.defer import Deferred, succeed, fail
+from twisted.trial.unittest import TestCase
+
+from benchmark.benchmark_driver import benchmark
+from benchmark.benchmark_operations import IProbe, IOperation
+
+
+class FakeMeasurement:
+
+    def __init__(self, measurements):
+        """
+        :param measurements: An iterator providing measurement to be
+            returned on each call.
+        """
+        self.measurements = measurements
+
+    def __call__(self, f, *a, **kw):
+        def finished(ignored):
+            return next(self.measurements)
+
+        d = f(*a, **kw)
+        d.addCallback(finished)
+        return d
+
+
+@implementer(IProbe)
+class FakeProbe:
+    """
+    A probe performs a single operation, which can be timed.
+    """
+    def __init__(self, good):
+        self.good = good
+
+    def run(self):
+        if self.good:
+            return succeed(None)
+        else:
+            return fail(RuntimeError('ProbeFailed'))
+
+    def cleanup(self):
+        return succeed(None)
+
+
+@implementer(IOperation)
+class FakeOperation:
+
+    def __init__(self, succeeds):
+        """
+        :param succeeds: An iterator providing a boolean indicating
+            whether the probe will succeed.
+        """
+        self.succeeds = succeeds
+
+    def get_probe(self):
+        return FakeProbe(next(self.succeeds))
```

Comment:
If this pair were to go into `operations`, then I would suggest splitting `FakeProbe` in two: a trivial succeeding probe and a failing probe. `FakeOperation` would then take an iterable of probes, and `get_probe` would just return whatever was necessary.

I think that's cleaner than passing a boolean around, and so it might be worthwhile doing here. What's here is fine though.

:grey_question: